src/AutoRest.CSharp/Output/Models/Requests/MgmtLongRunningOperation.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Output.Builders;
+using AutoRest.CSharp.Output.Models.Serialization;
+using AutoRest.CSharp.Output.Models.Types;
+using Azure;
+using Azure.Core;
+
+namespace AutoRest.CSharp.Output.Models.Requests
+{
+    internal class MgmtLongRunningOperation: TypeProvider
+    {
+        public MgmtLongRunningOperation(OperationGroup operationGroup, Input.Operation operation, BuildContext<MgmtOutputLibrary> context) : base(context)
+        {
+            Debug.Assert(operation.IsLongRunning);
+
+            var clientClass = context.Library.FindClient(operationGroup);
+
+            Debug.Assert(clientClass != null, "clientClass != null, LROs should be disabled when public clients are disables");
+
+            Client = clientClass;
+            DefaultName = clientClass.RestClient.ClientPrefix + operation.CSharpName() + "Operation";
+            FinalStateVia = operation.LongRunningFinalStateVia switch
```

Comment:
This is a lot of copied code.

Reply:
I was following the pattern, sharing code in most of these places requires more templatization which didn't seem like a route that was desirable. If we do this we can do it in a later PR

Reply:
I would also be happy to see a prototype that uses templatization.

Reply:
You could also use a builder and provide filter or before/after delegates to customize output.

I can help spike something here if you aren't tracking.

Reply:
There is only 1 thing that is library specific in this entire implementation. 

```
                 NextPageMethod = Client.RestClient.GetNextOperationMethod(operation.Requests.Single());

```

we can make it virtual. The Client property doesn't seem to be used at all so that can be deleted.

Reply:
Long-story short I think all of this logic can go into the base type.